### ISSUE TITLE
TypeScript module

### ISSUE THREAD

**Issue by mottyc** [stoicflame/enunciate#941]
Is there any development for TypeScript module?
If I'd like to write one, where is the best way to start? javascript / C# module?

**Comment by stoicflame** [comment]
Cool.

Yes, I'd suggest just copying whatever existing module might be closest to what you're looking for. I assume you'd start with JSON and not XML?

**Comment by mottyc** [comment]
Yes, I'd like to create a TypeScript module (single file) which includes all the data types and service methods of the API. I'm looking for the closest module that I can reuse (preferably only changing FreeMarker templates). Should it be an external module?

**Comment by stoicflame** [comment]
If you send a pull request, I'd be happy to take a look and consider merging it. Some people prefer to keep their module separate so they have more control over bug fixes and release schedules. 

Note that there is also a `javascript` module you may want to consider as the base.

**Comment by sortel64** [comment]
mottyc, How is this coming along?  I would be very interesting in this as well!  Is there anything I can do to help?  Thanks.

**Comment by agaengel** [comment]
we use Enunciate in our APIs and have https://github.com/vojtechhabarta/typescript-generator as additional plugin to generate a TypeScript client.

This could also be a solution to start with.

Regards
Alex
